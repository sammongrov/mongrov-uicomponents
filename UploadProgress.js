import React, { Component } from 'react';
import { Text, View, Platform, ProgressBarAndroid, ProgressViewIOS } from 'react-native';
import PropTypes from 'prop-types';
import { iOSColors } from 'react-native-typography';

export default class UploadProgress extends Component {
  render() {
    const { uploadFilePercent } = this.props;
    const progressPercentage = uploadFilePercent ? Number(uploadFilePercent.toFixed(1)) : 0;
    const displayPercentage = uploadFilePercent ? `${Math.ceil(uploadFilePercent * 100)}%` : '0%';
    let progressBar;
    if (Platform.OS === 'ios') {
      progressBar = (
        <ProgressViewIOS
          progress={progressPercentage}
          progressTintColor={iOSColors.blue}
          trackTintColor="#A6B9D8"
          progressViewStyle="bar"
          style={{ width: '60%' }}
        />
      );
    } else {
      progressBar = (
        <ProgressBarAndroid
          styleAttr="Horizontal"
          indeterminate={false}
          color={iOSColors.blue}
          progress={progressPercentage}
          style={{ width: '60%' }}
        />
      );
    }
    return (
      <View
        style={{
          position: 'absolute',
          width: '100%',
          top: '45%',
          left: 0,
          right: 0,
          flex: 1,
          alignItems: 'center',
        }}
      >
        {/* <Text style={{ color: iOSColors.gray, padding: 5 }}>Sending image...</Text> */}
        <Text style={{ color: iOSColors.blue }}>{displayPercentage}</Text>
        {progressBar}
      </View>
    );
  }
}

UploadProgress.propTypes = {
  uploadFilePercent: PropTypes.number,
};

UploadProgress.defaultProps = {
  uploadFilePercent: 0,
};
