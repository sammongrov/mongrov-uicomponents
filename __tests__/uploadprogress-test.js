import React from 'react';
import renderer from 'react-test-renderer';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import UploadProgress from '../UploadProgress';

configure({ adapter: new Adapter() });
jest.mock('react-native-router-flux', () => ({
  Actions: { pop: jest.fn() },
}));

beforeEach(() => {
  jest.resetModules();
});

it('render correctly without props', () => {
  jest.doMock('Platform', () => {
    const Platform = require.requireActual('Platform');
    Platform.OS = 'android';
    return Platform;
  });
  const rootComponent = renderer.create(<UploadProgress />).toJSON();
  expect(rootComponent).toMatchSnapshot();
});

it('android', () => {
  jest.doMock('Platform', () => {
    const Platform = require.requireActual('Platform');
    Platform.OS = 'android';
    return Platform;
  });
  const tree = renderer.create(<UploadProgress uploadFilePercent={0.3281} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('ios', () => {
  jest.doMock('Platform', () => {
    const Platform = require.requireActual('Platform');
    Platform.OS = 'ios';
    return Platform;
  });
  const tree = renderer.create(<UploadProgress uploadFilePercent={0.98756} />).toJSON();
  expect(tree).toMatchSnapshot();
});
