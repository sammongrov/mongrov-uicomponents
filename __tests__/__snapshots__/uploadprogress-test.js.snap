// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`android 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": "45%",
      "width": "100%",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#007AFF",
      }
    }
  >
    33%
  </Text>
  <View
    style={
      Array [
        Object {
          "alignSelf": "flex-start",
          "borderColor": "red",
          "borderWidth": 1,
        },
        Object {
          "width": "60%",
        },
      ]
    }
  />
</View>
`;

exports[`ios 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": "45%",
      "width": "100%",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#007AFF",
      }
    }
  >
    99%
  </Text>
  <RCTProgressView
    progress={1}
    progressTintColor="#007AFF"
    progressViewStyle="bar"
    style={
      Array [
        Object {
          "height": 2,
        },
        Object {
          "width": "60%",
        },
      ]
    }
    trackTintColor="#A6B9D8"
  />
</View>
`;

exports[`render correctly without props 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": "45%",
      "width": "100%",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#007AFF",
      }
    }
  >
    0%
  </Text>
  <View
    style={
      Array [
        Object {
          "alignSelf": "flex-start",
          "borderColor": "red",
          "borderWidth": 1,
        },
        Object {
          "width": "60%",
        },
      ]
    }
  />
</View>
`;
